packages/components/src/utils/rtl.js

```diff
@@ -20,18 +25,35 @@ export function useRtl() {
  * @param {Object} ltrStyles
  * @return {Object} Converted ltr -> rtl styles
  */
-const convertLtrToRtl = ( ltrStyles = {} ) => {
+export const convertLtrToRtl = ( ltrStyles = {} ) => {
 	const nextStyles = {};
 
 	for ( const key in ltrStyles ) {
 		const value = ltrStyles[ key ];
 		let nextKey = key;
-		if ( /left/gi.test( key ) ) {
-			nextKey = [ key.replace( 'left', 'right' ) ];
+
+		// Direct flip
+		if ( key === 'left' ) {
+			nextKey = 'right';
+		}
+		if ( key === 'right' ) {
+			nextKey = 'left';
+		}
+		// Lowercase flip
+		if ( lowerLeftRegExp.test( key ) ) {
+			nextKey = key.replace( lowerLeftRegExp, '-right' );
 		}
```

Comment:
Per [my reference comment](https://github.com/WordPress/gutenberg/pull/19916/files/110831f8f61f5e43ff9c76331159415040b412c9#r379631860), would there be any difference if this was implemented as:

```suggestion
		nextKey = key.replace( lowerLeftRegExp, '-right' );
```

Reply:
@aduth I just tried. It breaks one of the conversions in the unit tests

```
  convertLtrToRtl
    ✓ converts (*)Left <-> (*)Right (6ms)
    ✕ converts (*)left <-> (*)right (7ms)

  ● convertLtrToRtl › converts (*)left <-> (*)right

    expect(received).toBe(expected) // Object.is equality

    Expected: 10
    Received: 20

      105 | 		expect( nextStyle[ 'scroll-margin-right' ] ).toBe( 10 );
      106 | 		expect( nextStyle[ 'scroll-padding-right' ] ).toBe( 10 );
    > 107 | 		expect( nextStyle.right ).toBe( 10 );
          | 		                          ^
      108 |
      109 | 		// right -> left
      110 | 		expect( nextStyle[ 'border-left' ] ).toBe( '20px solid blue' );

      at Object.toBe (packages/components/src/utils/test/rtl.js:107:29)
```

Reply:
I think the reason is that we should be doing the replace as operating on the working copy of `nextKey`.

```suggestion
		nextKey = nextKey.replace( lowerLeftRegExp, '-right' );
```